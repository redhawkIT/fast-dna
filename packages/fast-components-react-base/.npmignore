# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
coverage/
src/
www/

# babel config
.babelrc
tsconfig.json
tslint.json
webpack.config.js