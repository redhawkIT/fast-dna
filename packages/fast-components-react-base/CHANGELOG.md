# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.1.0"></a>
# 1.1.0 (2018-05-09)


### Bug Fixes

* ensure app build and tslint processes run prior in the build gate ([#132](https://github.com/Microsoft/fast-dna/issues/132)) ([e74f953](https://github.com/Microsoft/fast-dna/commit/e74f953))
* **naming:** fix component names to align to repo standard ([#202](https://github.com/Microsoft/fast-dna/issues/202)) ([298836e](https://github.com/Microsoft/fast-dna/commit/298836e))
* **tslint:** fixes incorrect tslint rule regarding ordered imports ([#188](https://github.com/Microsoft/fast-dna/issues/188)) ([ebe0b30](https://github.com/Microsoft/fast-dna/commit/ebe0b30))
* fix tslint globbing issue and enforce whitespace in import/export statements ([#219](https://github.com/Microsoft/fast-dna/issues/219)) ([4637a90](https://github.com/Microsoft/fast-dna/commit/4637a90))
* travis-CI build-break ([#336](https://github.com/Microsoft/fast-dna/issues/336)) ([bffbf5e](https://github.com/Microsoft/fast-dna/commit/bffbf5e))


### Features

* add divider component ([#205](https://github.com/Microsoft/fast-dna/issues/205)) ([ae25c38](https://github.com/Microsoft/fast-dna/commit/ae25c38))
* add form generator to the packages ([#311](https://github.com/Microsoft/fast-dna/issues/311)) ([a339b3c](https://github.com/Microsoft/fast-dna/commit/a339b3c))
* add hypertext component ([#210](https://github.com/Microsoft/fast-dna/issues/210)) ([9e363ff](https://github.com/Microsoft/fast-dna/commit/9e363ff))
* add snapshot test suite ([#207](https://github.com/Microsoft/fast-dna/issues/207)) ([7ceaafe](https://github.com/Microsoft/fast-dna/commit/7ceaafe))
* catagorizing relevant dependencies as peerDependencies ([#186](https://github.com/Microsoft/fast-dna/issues/186)) ([7e15db6](https://github.com/Microsoft/fast-dna/commit/7e15db6))
* Forked Class name contracts so we can have one for Base and one for MSFT ([#262](https://github.com/Microsoft/fast-dna/issues/262)) ([a4c54c0](https://github.com/Microsoft/fast-dna/commit/a4c54c0))
* **typography:** add typography as a new base component and style ([#247](https://github.com/Microsoft/fast-dna/issues/247)) ([df3804e](https://github.com/Microsoft/fast-dna/commit/df3804e))
* remove JSS manager dependency from React base components ([#148](https://github.com/Microsoft/fast-dna/issues/148)) ([48de34a](https://github.com/Microsoft/fast-dna/commit/48de34a))
* **button:** updates to current msft styles ([#314](https://github.com/Microsoft/fast-dna/issues/314)) ([0029e06](https://github.com/Microsoft/fast-dna/commit/0029e06))
* update code coverage on travis ([#330](https://github.com/Microsoft/fast-dna/issues/330)) ([63ab4f4](https://github.com/Microsoft/fast-dna/commit/63ab4f4))
* **checkbox:** add new component with styles ([#252](https://github.com/Microsoft/fast-dna/issues/252)) ([3ad3988](https://github.com/Microsoft/fast-dna/commit/3ad3988))
* **Image:** add new component and msft styles ([#237](https://github.com/Microsoft/fast-dna/issues/237)) ([ea057ed](https://github.com/Microsoft/fast-dna/commit/ea057ed))
* **label:** add new component and msft styles ([#265](https://github.com/Microsoft/fast-dna/issues/265)) ([0328028](https://github.com/Microsoft/fast-dna/commit/0328028))
* **structure:** updates the structure for the development site ([#217](https://github.com/Microsoft/fast-dna/issues/217)) ([e153b0b](https://github.com/Microsoft/fast-dna/commit/e153b0b))
* **toggle:** add new component and msft styles ([#212](https://github.com/Microsoft/fast-dna/issues/212)) ([b9dd3e0](https://github.com/Microsoft/fast-dna/commit/b9dd3e0))
* update to React 16.3 ([#251](https://github.com/Microsoft/fast-dna/issues/251)) ([1fe77ef](https://github.com/Microsoft/fast-dna/commit/1fe77ef))
