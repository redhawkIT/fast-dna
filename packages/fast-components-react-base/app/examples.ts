import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
