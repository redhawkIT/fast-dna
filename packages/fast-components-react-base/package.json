{
  "name": "@microsoft/fast-components-react-base",
  "description": "FAST base components implemented in React.",
  "version": "1.0.4",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "clean:dist": "node ../../build/clean.js dist",
    "build:app": "webpack --progress",
    "prepare": "npm run clean:dist && tsc -p ./tsconfig.json",
    "start": "webpack-dev-server",
    "test": "npm run tslint && npm run unit-tests && npm run build:app",
    "tslint": "tslint -c ./tslint.json 'src/**/*.ts{,x}' '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix 'src/**/*.ts{,x}' '**/*.ts{,x}'",
    "unit-tests": "jest",
    "unit-tests:update": "jest --updateSnapshot",
    "unit-tests:watch": "jest --watch"
  },
  "license": "MIT",
  "readme": "README.md",
  "jest": {
    "collectCoverage": true,
    "transform": {
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.jsx?$": "babel-jest"
    },
    "transformIgnorePatterns": [
      "!<rootDir>/node_modules/lodash-es"
    ],
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx"
    ]
  },
  "devDependencies": {
    "@microsoft/fast-development-site-react": "1.0.4",
    "@microsoft/fast-jest-snapshots-react": "^1.0.4",
    "@microsoft/fast-tslint-rules": "^1.0.4",
    "@types/jest": "^22.2.0",
    "@types/lodash-es": "^4.17.0",
    "@types/node": "^9.4.6",
    "@types/react": "^16.3.0",
    "babel-core": "^6.26.0",
    "babel-jest": "^22.4.1",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "glob": "^7.1.2",
    "html-webpack-plugin": "^3.0.4",
    "jest": "^22.4.2",
    "lodash-es": "^4.0.0",
    "react": "^16.3.0",
    "react-dom": "^16.3.0",
    "regenerator-runtime": "^0.11.1",
    "rimraf": "^2.6.2",
    "ts-jest": "^22.4.1",
    "ts-loader": "^4.0.1",
    "tslint": "^5.9.1",
    "tslint-loader": "^3.6.0",
    "typescript": "^2.7.2",
    "webpack": "^4.1.0",
    "webpack-cli": "^2.0.11",
    "webpack-dev-server": "^3.1.0",
    "yargs": "^11.0.0"
  },
  "peerDependencies": {
    "lodash-es": "^4.0.0",
    "react": "^16.3.0",
    "react-dom": "^16.3.0"
  },
  "dependencies": {
    "@microsoft/fast-components-class-name-contracts-base": "^1.0.4"
  }
}
