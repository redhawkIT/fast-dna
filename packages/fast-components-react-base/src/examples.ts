import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
