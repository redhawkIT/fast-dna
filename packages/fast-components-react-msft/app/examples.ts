import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
