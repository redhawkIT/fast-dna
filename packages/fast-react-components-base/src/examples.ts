import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}