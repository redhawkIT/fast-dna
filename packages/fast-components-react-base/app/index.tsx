import * as React from "react";
import * as ReactDOM from "react-dom";
import Site, {
    componentFactory,
    ISiteProps,
    SiteCategory,
    SiteCategoryIcon,
    SiteCategoryItem
} from "@microsoft/fast-development-site-react";
import * as examples from "./examples";

/**
 * Create the root node
 */
const root: HTMLElement = document.createElement("div");
root.setAttribute("id", "root");
document.body.appendChild(root);

/* tslint:disable */
function render(): void {
    ReactDOM.render(
        <Site title={"FAST components base"}>
            <SiteCategory slot={"category"} name={"Components"}>
                {componentFactory(examples)}
            </SiteCategory>
        </Site>,
        root
    );
}

render();
