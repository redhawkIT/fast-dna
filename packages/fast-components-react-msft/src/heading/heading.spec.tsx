import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTHeading, {
    HeadingProps,
    HeadingSize,
    HeadingTag,
    IHeadingHandledProps,
    IHeadingManagedClasses,
    IHeadingUnhandledProps
} from "./heading";
import { Typography } from "@microsoft/fast-components-react-base";
import { Heading } from "./index";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("heading snapshots", (): void => {
    generateSnapshots(examples);
});

describe("heading", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTHeading as any).name).toBe(MSFTHeading.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(
            () => {
                shallow(<MSFTHeading tag={HeadingTag.h1} />);
                shallow(<MSFTHeading tag={HeadingTag.h1} size={HeadingSize._1} />);
            }
        ).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IHeadingHandledProps = {
            tag: HeadingTag.h1,
            size: HeadingSize._1
        };

        const unhandledProps: IHeadingUnhandledProps = {
            "aria-hidden": true
        };

        const props: IHeadingHandledProps & IHeadingUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Heading {...props} />
        );

        const heading: any = rendered.first().shallow();

        expect(heading.prop("aria-hidden")).toEqual(true);
    });

    test("should render the correct `tag` when `tag` prop is passed", () => {
        const rendered: any = shallow(
            <Heading tag={HeadingTag.h3} />
        );
        const heading: any = rendered.first().shallow();

        expect(heading.instance().props.tag).toEqual(HeadingTag.h3);
    });
});
