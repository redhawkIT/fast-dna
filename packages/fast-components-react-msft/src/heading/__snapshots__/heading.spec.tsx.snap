// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading snapshots Heading: 0 1`] = `
<h1
  className="typography-0-1-1 typography__1-0-1-2 heading-0-1-11 heading__1-0-1-12"
>
  Heading test string
</h1>
`;

exports[`heading snapshots Heading: 1 1`] = `
<h1
  className="typography-0-1-1 typography__1-0-1-2 heading-0-1-11 heading__1-0-1-12"
  data-sketch-symbol="Heading 1"
>
  Heading test string
</h1>
`;

exports[`heading snapshots Heading: 2 1`] = `
<h2
  className="typography-0-1-1 typography__2-0-1-3 heading-0-1-11 heading__2-0-1-13"
  data-sketch-symbol="Heading 2"
>
  Heading test string
</h2>
`;

exports[`heading snapshots Heading: 3 1`] = `
<h3
  className="typography-0-1-1 typography__3-0-1-4 heading-0-1-11 heading__3-0-1-14"
  data-sketch-symbol="Heading 3"
>
  Heading test string
</h3>
`;

exports[`heading snapshots Heading: 4 1`] = `
<h4
  className="typography-0-1-1 typography__4-0-1-5 heading-0-1-11 heading__4-0-1-15"
  data-sketch-symbol="Heading 4"
>
  Heading test string
</h4>
`;

exports[`heading snapshots Heading: 5 1`] = `
<h5
  className="typography-0-1-1 typography__5-0-1-6 heading-0-1-11 heading__5-0-1-16"
  data-sketch-symbol="Heading 5"
>
  Heading test string
</h5>
`;

exports[`heading snapshots Heading: 6 1`] = `
<h6
  className="typography-0-1-1 typography__6-0-1-7 heading-0-1-11 heading__6-0-1-17"
  data-sketch-symbol="Heading 6"
>
  Heading test string
</h6>
`;
